core/src/main/java/aima/core/search/api/BidirectionalProblem.java

```diff
@@ -0,0 +1,16 @@
+package aima.core.search.api;
+
+import java.util.List;
+
+/**
+ * An interface describing a problem that can be tackled from both directions at
+ * once (i.e InitialState<->Goal).
+ *
+ * @author Ciaran O'Reilly
+ *
+ */
+public interface BidirectionalProblem<A, S> {
+  Problem<A, S> getOriginalProblem();
+
+  List<Problem<A, S>> getReverseProblems();
+}
```

Comment:
Bidirectional implies functioning in two directions. This could be > 2, which is not what we want. It should be Problem<A, S> getReverseProblem().